src/main/java/com/gmail/chernii/oleksii/dao/DAO.java

```diff
@@ -0,0 +1,14 @@
+package com.gmail.chernii.oleksii.dao;
+
+/**
+ * Created by Space on 16.04.2019.
+ */
+public interface DAO<T> {
```

Comment:
It is better to create a specific interface for each dao. For example:
```
public interface CustomerDAO extends Dao<Customer>{
// some specific methods
} 
```
And after that implement this interface.